src/support/err.c

```diff
@@ -340,6 +342,7 @@ __eventv(WT_SESSION_IMPL *session, bool is_json, int error, const char *func, in
              */
             no_stderr = true;
             WT_ERR(__wt_vsnprintf_len_set(tmp->mem, tmp->memsize, &len, fmt, ap_copy));
+            WT_ERR(__wt_session_set_last_error(wt_session, error, WT_NONE, tmp->mem));
```

Comment:
My preference is to set the error towards the end of the function instead multiple place here.

Reply:
Done for the json case in [Call set_last_error only once after var-arg formatting for json](https://github.com/wiredtiger/wiredtiger/pull/11361/commits/c605e7cc6b7671e9c447ba9d7e60f47af3d4d4ea), it's more difficult for the non-json case since the buffer is immediately altered (`p += len`) after being written to.